corpusreaders/src/main/java/edu/illinois/cs/cogcomp/nlp/corpusreaders/ereReader/EREDocumentReader.java

```diff
@@ -7,42 +7,162 @@
  */
 package edu.illinois.cs.cogcomp.nlp.corpusreaders.ereReader;
 
-import java.util.ArrayList;
+import java.io.FilenameFilter;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.*;
 
-import edu.illinois.cs.cogcomp.core.io.LineIO;
+import edu.illinois.cs.cogcomp.core.io.IOUtils;
 import edu.illinois.cs.cogcomp.core.utilities.TextCleaner;
+import edu.illinois.cs.cogcomp.nlp.corpusreaders.ACEReader;
 import edu.illinois.cs.cogcomp.nlp.corpusreaders.XmlFragmentWhitespacingDocumentReader;
 
+import static edu.illinois.cs.cogcomp.core.io.IOUtils.getFileName;
+import static edu.illinois.cs.cogcomp.core.io.IOUtils.getFileStem;
+
 /**
  * Strips all useless XML markup from an ERE xml document leaving the original text and 
  * where needed, appropriate attribute values.
- * @author redman
+ * Base class for other ERE readers for Named Entities, Mentions/Relations, and Events.
  *
+ * @author redman
+ * @author msammon
  */
 public class EREDocumentReader extends XmlFragmentWhitespacingDocumentReader {
-	
-	/** these tags contain attributes we want to keep. */
-    static private ArrayList<String> retainTags = new ArrayList<String>();
+
+    /**
+     * tags in ERE markup filess
+     */
+    public static final String ENTITIES = "entities";
+    public static final String ENTITY = "entity";
+    public static final String FILLERS = "fillers";
+    public static final String FILLER = "filler";
+    public static final String OFFSET = "offset";
+    public static final String TYPE = "type";
+    public static final String ENTITY_MENTION = "entity_mention";
+    public static final String NOUN_TYPE = "noun_type";
+    public static final String PRO = "PRO";
+    public static final String NOM = "NOM";
+    public static final String NAM = "NAM";
+    public static final String FILL = "FILL";
+    public static final String LENGTH = "length";
+    public static final String MENTION_TEXT = "mention_text";
+    public static final String MENTION_HEAD = "nom_head";
+    public static final String SPECIFICITY = "specificity";
+    public static final String REALIS = "realis";
+    public static final String ID = "id";
+    public static final String RELATIONS = "relations";
+    public static final String RELATION = "relation";
+    public static final String RELATION_MENTION = "relation_mention";
+    public static final String SUBTYPE = "subtype";
+    public static final String ARG_ONE = "rel_arg1";
+    public static final String ARG_TWO = "rel_arg2";
+    public static final String ENTITY_MENTION_ID = "entity_mention_id";
+    public static final String ENTITY_ID = "entity_id";
+    public static final String ROLE = "role";
+    public static final String FILLER_ID = "filler_id";
+
+
+
+    /** aim for consistent naming */
```

Comment:
This is a vexed issue, because different corpora almost invariably have overlapping but non-identical tag sets.  I opted here to have a local copy of all the variables, and to reuse those where they were the same in an attempt to help anyone who is moving between ERE and ACE to identify where the tags and attributes are the same and where they are different -- it affects what feature extractors can be reused, for example.